examples/stable_diffusion_v2/cldm/cldm.py

```diff
@@ -355,25 +360,44 @@ def __init__(self, control_stage_config, control_key, only_mid_control, *args, *
         self.only_mid_control = only_mid_control
         self.control_scales = [1.0] * 13
 
-    def get_input(self, batch, k, bs=None, *args, **kwargs):
-        self.set_train(False)
-        x, c = super().get_input(batch, self.first_stage_key, *args, **kwargs)
-        control = batch[self.control_key]
+    def construct(self, x, c, control):
+        t = self.uniform_int(
+            (x.shape[0],), ms.Tensor(0, dtype=ms.dtype.int32), ms.Tensor(self.num_timesteps, dtype=ms.dtype.int32)
+        )
+        # x, c_crossattn, c_concat = self.get_input(x, c, control)
+        # batch: images, captions, controls
+        if len(x.shape) == 3:
+            x = x[..., None]
+        x = self.transpose(x, (0, 3, 1, 2))
+        z = ops.stop_gradient(self.get_first_stage_encoding(self.encode_first_stage(x)))
+        c = ops.stop_gradient(self.cond_stage_model.encode(c))
+        control = ops.transpose(control, (0, 3, 1, 2))  # 'b h w c -> b c h w'
+        x, c_crossattn, c_concat = z, c, control
```

Comment:
Replace this with `get_input()` as these parts are identical?

Reply:
Just try to accelerate.

Reply:
I think if you call a function inside `construct()`, that function will also be compiled as a graph and won't affect execution speed.